examples/server/server.cpp

```diff
@@ -2841,7 +2866,9 @@ struct server_context {
                 for (size_t i = 0; i < ids.size(); ++i) {
                     completion_token_output result;
 
-                    result.tok = ids[i];
+                    result.tok          = ids[i];
+                    result.text_to_send = common_token_to_piece(ctx, result.tok, params_base.special);
+                    result.prob         = 1.0f; // set later
```

Comment:
Here is the branch for speculative decoding. I'm not sure now I can get token probs here. Could you give me some clues? @ggerganov

Reply:
Think we need to update `common_sampler_sample_and_accept_n()` to return the probs. But let's fix this later.